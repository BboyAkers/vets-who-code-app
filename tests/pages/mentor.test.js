import React from 'react'
import Mentor from '../../src/pages/mentor'


describe('<Mentor />', () => {
  let wrapper

  beforeEach(() => wrapper = shallow(<Mentor />))

  test('should update state handleChange is invoked', () => {
    const mockEvent = {
      target: {
        name: 'name',
        value: 'User Name'
      }
    }
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('name')).toEqual('User Name')
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
