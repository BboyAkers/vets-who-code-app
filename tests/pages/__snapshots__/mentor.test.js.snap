// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mentor /> should render correctly 1`] = `
<TemplateWrapper>
  <header
    className="inner-header overlay grey text-center slim-bg "
    style={
      Object {
        "backgroundImage": "url(image-mock)",
        "backgroundPositionY": "bottom",
      }
    }
  >
    <div
      className="overlay-01"
    />
    <div
      className="container"
    >
      <h2
        className="text-center text-uppercase"
      >
        Mentor
      </h2>
      <div
        className="breadcrumb"
      >
        <Link
          to="/"
        >
          Home
        </Link>
        <span>
          /
        </span>
        <Link
          className="page-active"
          to="/mentor"
        >
          Mentor
        </Link>
      </div>
    </div>
  </header>
  <section
    className="pad-regular section bg-default"
    id="contact"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-12"
        >
          <div
            className="contactus-brief"
          >
            <h3>
              Become A Mentor
            </h3>
            <p
              className="section-description"
            >
              Thank you for choosing to become a mentor for our veterans. Please fill out the form below and we will reach out to you.
            </p>
            <form
              id="s2do-form"
              onSubmit={[Function]}
            >
              <div
                className="col-md-8"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="dark-text"
                    htmlFor="InputName"
                  >
                    Name
                    <sup>
                      *
                    </sup>
                  </label>
                  <input
                    className="form-control input-lg"
                    name="name"
                    onChange={[Function]}
                    placeholder="Name"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="dark-text"
                    htmlFor="InputEmail"
                  >
                    Email
                    <sup>
                      *
                    </sup>
                  </label>
                  <input
                    className="form-control input-lg"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required={true}
                    type="email"
                    value=""
                  />
                </div>
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="dark-text"
                    htmlFor="InputService"
                  >
                    Military Branch Of Service
                    <sup>
                      *
                    </sup>
                  </label>
                  <input
                    className="form-control input-lg"
                    name="branch-of-service"
                    onChange={[Function]}
                    placeholder="Thank you for your service"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="dark-text"
                    htmlFor="InputTechnicalExpertise"
                  >
                    Area Of Technical Expertise ( Javascript, Ruby, etc)
                    <sup>
                      *
                    </sup>
                  </label>
                  <input
                    className="form-control input-lg"
                    name="technical-expertise"
                    onChange={[Function]}
                    placeholder="What languages you write in?"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="dark-text"
                    htmlFor="InputPortfolio"
                  >
                    Github, Portfolio or Linkedin
                    <sup>
                      *
                    </sup>
                  </label>
                  <input
                    className="form-control input-lg"
                    name="github-portfolio-or-linkedin"
                    onChange={[Function]}
                    placeholder="Share your work"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="dark-text"
                    htmlFor="InputLocation"
                  >
                    Location ( City and State )
                    <sup>
                      *
                    </sup>
                  </label>
                  <input
                    className="form-control input-lg"
                    name="location"
                    onChange={[Function]}
                    placeholder="Location"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="dark-text"
                    htmlFor="InputEmployerImplementedRestricted"
                  >
                    Employer Implemented Restrictions, If Any.
                  </label>
                  <textarea
                    className="form-control"
                    name="employer-restrictions"
                    onChange={[Function]}
                    placeholder="Please put here any employer restrictions about writing and reading code?"
                    rows="7"
                    value=""
                  />
                </div>
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="form-group"
                >
                  <input
                    className="btn btn-charity-default"
                    href="#"
                    name="submit"
                    title=""
                    type="submit"
                  />
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </section>
</TemplateWrapper>
`;
